src/configure/utility/pipelineHelper.ts

```diff
@@ -0,0 +1,60 @@
+import * as fs from "fs";
+import * as path from "path";
+
+import * as vscode from 'vscode';
+import { PipelineTargets } from "../model/models";
+
+export async function analyzeRepoAndListAppropriatePipeline(repoPath: string): Promise<string[]> {
+    let fileUris = await analyzeRepo(repoPath);
+
+    // TO-DO: To populate the possible pipelines on the basis of azure target resource.
+    let appropriatePipelines: string[] = [];
+    if (fileUris.nodeUris.length > 0) {
+        appropriatePipelines.push(NodeOnWindows);
+    }
+    else if (fileUris.dockerUris.length > 0) {
+        appropriatePipelines.push(NodeOnContainers);
+    }
+    else {
+        appropriatePipelines.push(NodeOnWindows, NodeOnContainers);
```

Comment:
I think as per the spec, we show all the options but the order is as per the analysis of the repo. like show container option first if dockerfile is found. Do we want to NOT show certain pipeline options ?

Reply:
Taking it up with the PM, as it is not clear in the SPEC.

Reply:
Discussed it with PM. We shall be fine if we are not showing the one's which are not valid according to us. 
This helps in keeping the choices clear for the user.